pkg/infra/assembler.go

```diff
@@ -19,30 +21,38 @@ type Assembler struct {
 	Signer *Crypto
 }
 
-func (a *Assembler) assemble(e *Elements) *Elements {
+func (a *Assembler) assemble(e *Elements, logger *log.Logger) (*Elements, error) {
 	env, err := CreateSignedTx(e.Proposal, a.Signer, e.Responses)
 	if err != nil {
-		panic(err)
+		errCP := errors.Wrapf(err, "error sign tx")
+		logger.Error(errCP)
+		return nil, errCP
 	}
 	e.Envelope = env
-	return e
+	return e, nil
 }
 
-func (a *Assembler) sign(e *Elements) *Elements {
+func (a *Assembler) sign(e *Elements, logger *log.Logger) (*Elements, error) {
 	sprop, err := SignProposal(e.Proposal, a.Signer)
 	if err != nil {
-		panic(err)
+		errCP := errors.Wrapf(err, "error proposal")
+		logger.Error(errCP)
```

Comment:
in this case, error is returned and logged at root caller, there's no need to repeatedly log the same error

Reply:
k, will remove.